src/SOS/Strike/strike.cpp

```diff
@@ -1188,7 +1196,15 @@ DECLARE_API(DumpMT)
     DacpMethodTableCollectibleData vMethTableCollectible;
     vMethTableCollectible.Request(g_sos, TO_CDADDR(dwStartAddr));
 
-    table.WriteRow("EEClass:", EEClassPtr(vMethTable.Class));
+    // Since .NET 9, DacpMethodTableData:Class is the canonical method table for the given method table.
+    // We can check if the Class member is the same as the result of GetMethodTableForEEClass to determine if we're on .NET9+
+    BOOL net9PreferCanonMT = FALSE;
```

Comment:
nit: Why not just eliminate this variable and use PreferCanonMTOverEEClassForNET9 directly below?

Also instead of an `if` statement, you could do:  `bool net9PreferCanonMT = PreferCanonMTOverEEClassForNET9(vMethTable.Class, net9PreferCanonMT) == S_OK;` ?

Reply:
changed the helper method to return `S_OK`/`S_FALSE` instead of using an out bool.  I'm not sure it cleaned things up a ton.  In some places we still want to handle failure by propagating the HRESULT out

Reply:
I'm not sure I like the overloading of the return valid. Returning the flag explicitly seems clearer to me and if the function fails, the status just be returned in all places. This comment kind of negates my other comment about removing the net9PreferCaonMT flag.  

I'll leave it up to you now.

Reply:
I think I like it better with a separate flag.  This reads pretty idiomatic to me:

```cpp
  BOOL preferMT = FALSE;
  if (SUCCEEDED(PreferCanonMTOverEEClass(eeClassPtr, &preferMT)) && preferMT)
  {
     // do things with MethodTable
  }
  else
  {
     // do things with EEClass
  }
```

Reply:
I like that better.